frontend/javascripts/viewer/view/action-bar/starting_job_modals.tsx

```diff
@@ -712,28 +726,64 @@ function useCurrentlySelectedBoundingBox(
   return currentlySelectedBoundingBox;
 }
 
+// This function mirrors the selection of the mag
+// in voxelytics/worker/job_utils/voxelytics_utils.py select_mag_for_model_prediction
+const getBestFittingMagComparedToTrainingDS = (
+  colorLayer: APIDataLayer,
+  datasetScaleMag1: VoxelSize,
+  jobType: APIJobType.INFER_MITOCHONDRIA | APIJobType.INFER_NEURONS | APIJobType.INFER_NUCLEI,
+) => {
+  if (jobType === APIJobType.INFER_MITOCHONDRIA) {
+    // infer_mitochondria_model always infers on the finest mag of the current dataset
+    const magInfo = getMagInfo(colorLayer.resolutions);
+    return magInfo.getFinestMag();
+  }
+  const modelScale = MEAN_VX_SIZE[jobType];
+  let closestMagOfCurrentDS = colorLayer.resolutions[0];
+  let bestDifference = [
+    Number.POSITIVE_INFINITY,
+    Number.POSITIVE_INFINITY,
+    Number.POSITIVE_INFINITY,
+  ];
+
+  const datasetScaleInNm = convertVoxelSizeToUnit(datasetScaleMag1);
+
+  for (const mag of colorLayer.resolutions) {
+    const diff = datasetScaleInNm.map(
+      (dim, i) => Math.log2(dim * mag[i]) - Math.log2(modelScale[i]),
+    );
+    if (Math.abs(bestDifference[0]) > Math.abs(diff[0])) {
+      bestDifference = diff;
+      closestMagOfCurrentDS = mag;
+    }
+  }
+  const maxDistance = Math.max(...bestDifference);
```

Comment:
The original line is `np.max(np.abs(closest_diff))`. I think the `abs` is important because of negative differences and should be added here as well.

Reply:
I changed it so that the original calculation of the diff simply takes the absolute (l 750
`const diff = datasetScaleInNm.map((dim, i) => Math.abs(Math.log(dim * mag[i]) - Math.log(modelScale[i])));`. Then the diff is always positive and the absolute can only be taken once. This means that this code differs a bit from the vx code, but it does anyway because of the different ways numpy and TS handle arrays. If you think they should be as similar as possible, please let me know, then I will change it back.

Reply:
Great solution :+1: